plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -87,41 +87,11 @@ public function save_to_session( $customer ) {
 			return;
 		}
 
-		$data = array();
-		foreach ( $this->session_keys as $session_key ) {
-			$function_key = $session_key;
-			if ( 'billing_' === substr( $session_key, 0, 8 ) ) {
-				$session_key = str_replace( 'billing_', '', $session_key );
-			}
-			if ( 'meta_data' === $session_key ) {
-				/**
-				 * Filter the allowed session meta data keys.
-				 *
-				 * If the customer object contains any meta data with these keys, it will be stored within the WooCommerce session.
-				 *
-				 * @since 8.7.0
-				 * @param array $allowed_keys The allowed meta data keys.
-				 * @param WC_Customer $customer The customer object.
-				 */
-				$allowed_keys = apply_filters( 'woocommerce_customer_allowed_session_meta_keys', array(), $customer );
+		$data = $this->get_customer_session_data( $customer );
 
-				$session_value = array();
-				foreach ( $customer->get_meta_data() as $meta_data ) {
-					if ( in_array( $meta_data->key, $allowed_keys, true ) ) {
-						$session_value[] = array(
-							'key'   => $meta_data->key,
-							'value' => $meta_data->value,
-						);
-					}
-				}
-				$data['meta_data'] = $session_value;
-			} else {
-				$session_value        = $customer->{"get_$function_key"}( 'edit' );
-				$data[ $session_key ] = (string) $session_value;
-			}
+		if ( ! $this->is_default_customer_data( $data ) ) {
+			WC()->session->set( 'customer', $data );
```

Comment:
```suggestion
		if ( $this->is_default_customer_data( $data ) ) {
			// Clear the customer from the session if it matches the default.
			WC()->session->set( 'customer', null );
		} else {
			WC()->session->set( 'customer', $data );
		}
```

We can clear the customer from the session if it matches the default.

Reply:
Thanks, good suggestion, committing it!